File: frontend/src/types/whitelist.ts
export type WhitelistType = 'process' | 'domain' | 'ip' | 'registry';

export interface Whitelist {
  id: string;
  type: WhitelistType;
  value: string;
  description: string;
  is_regex: boolean;
  enabled: boolean;
  created_at: string;
  updated_at: string;
}

export interface WhitelistCreate {
  type: WhitelistType;
  value: string;
  description?: string;
  is_regex?: boolean;
  enabled?: boolean;
}

File: internal/models/whitelist.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// WhitelistType represents the type of whitelist entry
type WhitelistType string

const (
	WhitelistTypeProcess  WhitelistType = "process"
	WhitelistTypeDomain   WhitelistType = "domain"
	WhitelistTypeIP       WhitelistType = "ip"
	WhitelistTypeRegistry WhitelistType = "registry"
)

// Whitelist represents a whitelist entry
type Whitelist struct {
	ID          uuid.UUID     `json:"id"`
	Type        WhitelistType `json:"type"`
	Value       string        `json:"value"`
	Description string        `json:"description"`
	IsRegex     bool          `json:"is_regex"`
	Enabled     bool          `json:"enabled"`
	CreatedAt   time.Time     `json:"created_at"`
	UpdatedAt   time.Time     `json:"updated_at"`
}

// WhitelistCreate represents the request to create a whitelist entry
type WhitelistCreate struct {
	Type        WhitelistType `json:"type" binding:"required"`
	Value       string        `json:"value" binding:"required"`
	Description string        `json:"description"`
	IsRegex     bool          `json:"is_regex"`
	Enabled     bool          `json:"enabled"`
}

// WhitelistUpdate represents the request to update a whitelist entry
type WhitelistUpdate struct {
	Value       *string `json:"value"`
	Description *string `json:"description"`
	IsRegex     *bool   `json:"is_regex"`
	Enabled     *bool   `json:"enabled"`
}
